src/nisarqa/products/igram.py

```diff
@@ -15,26 +15,30 @@
 
 
 def verify_igram(
-    user_rncfg: Mapping[str, Mapping], product_type: str, verbose: bool = False
+    root_params: (
+        nisarqa.RIFGRootParamGroup
+        | nisarqa.RUNWRootParamGroup
+        | nisarqa.GUNWRootParamGroup
+    ),
+    product_type: str,
+    verbose: bool = False,
 ) -> None:
     """
-    Verify a RIFG, RUNW, or GUNW product per provided runconfig.
+    Perform verification checks and quality assurance on a NISAR InSAR product.
 
-    This is the main function for running the entire QA workflow for this
-    product. It will run based on the options supplied in the
-    input runconfig file.
-    The input runconfig file must follow the standard QA runconfig format
-    for this product. Run the command line command:
-            nisar_qa dumpconfig <product name>
-    to generate an example template with default parameters for this product.
+    This is the main function for running the entire QA workflow for one of
+    a NISAR RIFG, RUNW, or GUNW product.
+    It will run based on the options supplied in the input parameters.
 
     Parameters
     ----------
-    user_rncfg : nested dict
-        A dictionary whose structure matches this product's QA runconfig
-        YAML file and which contains the parameters needed to run its QA SAS.
+    root_params : nisarqa.RIFGRootParamGroup or nisarqa.RUNWRootParamGroup
+                        or nisarqa.GUNWRootParamGroup
+        Input parameters to run this QA SAS. The instance type
+        should correspond to the `product_type`.
```

Comment:
Alternatively, we could remove the `product_type` argument and just infer the product type from the type of `root_params`. Note that the docstring should document this behavior if you decide to make this change.

Currently, `product_type` is only an argument for the `verify_igram` and `verify_offset` functions, so this would be a step towards harmonizing their interfaces with the other `verify_*` functions.